### PR TITLE
Add upload source maps step to the release checklist

### PR DESCRIPTION
**Pull request by fluiddot**
**Related to https://github.com/wordpress-mobile/gutenberg-mobile/issues/6023.**

This new step will ensure that we store the source maps of each version in their associated GitHub release.

I noticed it's important to use the same source maps produced when generating the bundle. Hence, I also added a step to locate and download these files that are stored as artifacts of the Buildkite job. This way we'll ensure when symbolicating a stack trace that we use the correct source maps.

Here are a couple of screenshots to see the changes in action:

Screenshot 2023-08-04 at 17 31 17

<hr>

Screenshot 2023-08-04 at 17 30 56

### CODE REVIEW

File: Releasing.md [hunk 1]
```diff
@@ -95,9 +95,49 @@ For the body of the post, just copy this checklist and again replace all occurre
...
+<p>o Mark all 4 PRs ready for review and request review from the team. </p>
 <!-- /wp:paragraph -->
 
+<!-- wp:paragraph {"style":{"color":{"background":"#fcb900ba"}}} -->
+<p class="has-background" style="background-color:#fcb900ba">⚠️ In some release cases (like beta fixes), it's likely that the PRs could have conflicts with <code>trunk</code>. In this case, do not resolve merge conflicts by merging with <code>trunk</code> as this will introduce new and unexpected changes to the release. Instead, leave the conflicts until the release is integrated into the main apps, and then resolve the conflicts when merging the PRs back to <code>trunk</code>. Optionally, a second clone of the release branch can be created to verify the CI checks.</p>
+<!-- /wp:paragraph -->
```
Comment (fluiddot) [inline Releasing.md:103]: This change is unrelated to the goal of this PR. It's just an update to the styles and text, to specify that the reviews should be requested by the team, as we no longer have the role of release wrangler buddy.

File: Releasing.md [hunk 2]
```diff
@@ -95,7 +95,11 @@ For the body of the post, just copy this checklist and again replace all occurre
...
+<p>o Mark all 4 PRs ready for review and request review from the team. </p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph {"style":{"color":{"background":"#fcb900ba"}}} -->
+<p class="has-background" style="background-color:#fcb900ba">⚠️ In some release cases (like beta fixes), it's likely that the PRs could have conflicts with <code>trunk</code>. In this case, do not resolve merge conflicts by merging with <code>trunk</code> as this will introduce new and unexpected changes to the release. Instead, leave the conflicts until the release is integrated into the main apps, and then resolve the conflicts when merging the PRs back to <code>trunk</code>. Optionally, a second clone of the release branch can be created to verify the CI checks.</p>
```
Comment (fluiddot) [inline Releasing.md:102]: This change is unrelated to the goal of this PR. It's just an update to the styles and text, to specify that the reviews should be requested by the team, as we no longer have the role of release wrangler buddy.

File: Releasing.md [hunk 3]
```diff
@@ -159,14 +163,40 @@ For the body of the post, just copy this checklist and again replace all occurre
...
+<p>o Ensure that the bundle files are updated to include any changes to the release branch by running <code>npm run bundle</code> in gutenberg-mobile release branch and committing any changes.</p>
 <!-- /wp:paragraph -->
 
+<!-- wp:paragraph -->
+<p>o <a href="https://github.com/wordpress-mobile/gutenberg-mobile/releases/new?tag=vX.XX.X&amp;target=release/X.XX.X&amp;title=Release%20X.XX.X">Create a new gutenberg-mobile GitHub Release</a>. Include a list of changes in the Release description. Ensure the checkmark "Set as the latest release" is checked, and <strong>publish the release with the "Publish release" button.</strong></p>
+<!-- /wp:paragraph -->
 
 <!-- wp:paragraph -->
-<p>o <a href="https://github.com/wordpress-mobile/gutenberg-mobile/releases/new?tag=vX.XX.X&amp;target=release/X.XX.X&amp;title=Release%20X.XX.X">Create a new gutenberg-mobile GitHub Release</a>. Include a list of changes in the Release description. Ensure the checkmark next "Set as the latest release" is checked, and publish the release with the "Publish release" button.</p>
+<p>o Wait until all CI jobs for the published tag finish and succeed.</p>
```
Comment (fluiddot) [inline Releasing.md:174]: Although this step is implicit, I added it because it's important to wait for all jobs before updating the references in the host apps.

File: Releasing.md [hunk 4]
```diff
@@ -159,14 +163,40 @@ For the body of the post, just copy this checklist and again replace all occurre
...
 <!-- /wp:paragraph -->
 
+<!-- wp:paragraph -->
+<p>o Navigate to the Buildkite job that built the JS bundles (<code>Build JS Bundles</code>) for the published tag. Open the job and navigate to the "Artifacts" tab. Locate the composed source maps (they have file name <code>bundle/{platform}/App.composed.js.map</code>) and download them.</p>
```
Comment (fluiddot) [inline Releasing.md:178]: An example of this can be seen in [this Buildkite job](https://buildkite.com/automattic/gutenberg-mobile/builds/6751#0189c0b8-e728-4b32-8b2f-c814dcd7e387).

File: Releasing.md [hunk 5]
```diff
@@ -159,14 +163,40 @@ For the body of the post, just copy this checklist and again replace all occurre
...
+<!-- /wp:paragraph -->
+
+<!-- wp:group -->
+<div class="wp-block-group"><!-- wp:paragraph -->
+<p>o Navigate and edit the GitHub release. Attach the composed source maps to the release <em>(you can drag and drop the files in the "Attach binaries" drop area)</em>. Once they are uploaded, update the artifact's name following this format:</p>
```
Comment (fluiddot) [inline Releasing.md:183]: An example of this can be checked in https://github.com/wordpress-mobile/gutenberg-mobile/releases/tag/v1.100.2.

File: Releasing.md (hunk 3 above)
Comment (SiobhyB) [reply Releasing.md:174]: I agree that there's value in adding this step here. Similarly, in the `Create the Release` section, we have a couple of tasks that relate to waiting for the Android `Publish to S3` test before creating the Android build.

**Line 64:**

```
<strong>Note:</strong> You might want to wait a bit before confirming WPAndroid PR creation so gutenberg-mobile can have enough time to finish the <code>Build Android RN Bridge &amp; Publish to S3</code> job on CI which is needed by WPAndroid CI.
```

**Line 86:**

```
<p>o Verify the WPAndroid and iOS PR builds succeed. For WPAndroid, if the PR CI tasks include a 403 error related to an inability to resolve the <code>react-native-bridge</code> dependency, you must wait for the <code>Build Android RN Bridge &amp; Publish to S3</code> task to succeed in gutenberg-mobile and then restart the WPAndroid CI tasks.</p>
```

What do you think about simplifying and adding details about the `Publish to S3` requirement for iOS as part of this PR? I'm thinking we could perhaps remove the note at line 64 and update line 86 to something like the following:

```
<p>o Verify the WPAndroid and iOS PR builds succeed. For WPAndroid, if the PR CI tasks include a 403 error related to an inability to resolve the <code>react-native-bridge</code> dependency, you must wait for the <code>Build Android RN Bridge &amp; Publish to S3</code> task to succeed in gutenberg-mobile and then restart the WPAndroid CI tasks. Similarly, for iOS, you must wait for the <code>Build iOS RN XCFramework  &amp; Publish to S3</code> task to succeed.</p>
```

Comment (fluiddot) [reply Releasing.md:174]: > What do you think about simplifying and adding details about the `Publish to S3` requirement for iOS as part of this PR? I'm thinking we could perhaps remove the note at line 64 and update line 86 to something like the following:

Yeah, good idea! I'll apply this suggestion.

Comment (fluiddot) [reply Releasing.md:174]: Suggestion applied in https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/pull/143/commits/7f529f90cebe96032d6c2f3658302006e0836014.

**Review by SiobhyB — APPROVED** [review]
These changes look great, _and_ I'm grateful you caught a few styling issues along the way. 😍  Thank you!

Note, we should update the P2 pattern when this is merged.

**Comment by fluiddot** [comment]
> Note, we should update the P2 pattern when this is merged.

Ah, interesting, I didn't know we had a P2 pattern. I'll take a look, thanks!
